lib/mergeBuffers.js

```diff
@@ -41,8 +41,19 @@ function mergeBuffers(gltf, bufferId) {
                         bufferView.buffer = bufferId;
                     }
                 }
-                buffersToMerge.push(buffer.extras._pipeline.source);
-                lengthSoFar += buffer.extras._pipeline.source.length;
+
+                var sourceBuffer = buffer.extras._pipeline.source;
+                var currentBufferLength = sourceBuffer.length;
+                if (currentBufferLength % 4)

```

Comment:
Yep. One thing that the note doesn't say is that a float must also start on 4-byte boundary. My particular case had something like 102 bytes of shorts containing batch ids so every float array after it in the buffer was unable to be read. The C++ converter handles this, but during the processing the padding is thrown out, so the problem creeps back in again.